### PR TITLE
GET Api/users

### PR DESCRIPTION
**Pull request by ji0kim**
GET /api/users
- 200 : success
- 404 : not found

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -163,4 +163,33 @@ describe('PATCH /api/articles/article_id', () => {
...
 			});
 	});
+});
+describe('GET /api/users', () => {
+	test('200 - Success : Get users from the users table', () => {
+		return request(app)
+			.get('/api/users')
+			.expect(200)
+			.then((response) => {
+				expect(response.body.users).toBeInstanceOf(Array);
```
Comment (karlaferry) [inline __tests__/app.test.js:173]: You want to check the length as well!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -163,4 +163,33 @@ describe('PATCH /api/articles/article_id', () => {
...
+				expect(response.body.users).toEqual([
+					{ username: 'butter_bridge' },
```
Comment (karlaferry) [inline __tests__/app.test.js]: Your individual user objects should have the following keys: `username`, `avatar_url`, and `name`.

File: models/users.model.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const db = require('../db/connection');
+
+exports.selectUsers = () => {
+	return db.query('SELECT * FROM users').then((result) => {
+		const users = result.rows.map((user) => {
```
Comment (karlaferry) [inline models/users.model.js]: No need to map! You need all the properties you get. 😄

File: controllers/users.controller.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const { selectUsers } = require('../models/users.model.js');
+exports.getUsers = (req, res) => {
+	selectUsers().then((users) => {
+		res.send({ users });
```
Comment (karlaferry) [inline controllers/users.controller.js]: Don't forget your res.status!

**Review by karlaferry — CHANGES_REQUESTED** [review]
Nice work! Please see my comments for the suggested improvements 😄

File: models/users.model.js (hunk 3 above)
Comment (ji0kim) [reply models/users.model.js]: I thought I only need the key of 'username'...!🥲

**Review by dentednerd — APPROVED** [review]
Looking good!

File: __tests__/app.test.js (hunk 2 above)
Comment (karlaferry) [reply __tests__/app.test.js]: ^^ Ignore my previous comment! 😄

File: models/users.model.js (hunk 3 above)
Comment (karlaferry) [reply models/users.model.js]: Yes you are right! Sorry just saw this.

**Pull request merged**